### PR TITLE
#161806791 Add get comments for an article feature

### PR DESCRIPTION
**Pull request by TheoOkafor**
#### What does this PR do?
- Add GET all comments and replies on an article

#### Description of Task to be completed?
- Create the route for getting comments and replies on an article
- Add tests for route
- write documentation

#### How should this be manually tested?
- git clone `https://github.com/andela/haven-ah-backend.git`.
- Install `node` if not installed.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.

To test on postman
- run `npm run start-dev`
- On postman run GET `localhost:5000/api/articles/:slug/comments` (set JWT token as`x-access-token` in header).
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
 #161806791

#### Screenshots (if appropriate)

Good Response 1:
screen shot 2018-11-12 at 4 01 42 pm
screen shot 2018-11-12 at 4 02 20 pm

### CODE REVIEW

File: server/controllers/comment.js [hunk 1]
```diff
@@ -117,6 +120,35 @@ class Comment {
...
+  static async getComments(request, response) {
+    const { slug } = request.params;
+
+    const article = await articleRepo.getArticleBySlug(slug);
+    if (article === null) {
+      return badHttpResponse(response, 404, 'We could not find this article');
+    }
+
+    const rawComments = await commentRepo.getCommentsOnArticle(article.id);
+    const maxParentId = Math.max(...rawComments.map(element => element.parentId));
```
Comment (victor-nweze-deel) [inline server/controllers/comment.js:135]: Why are we spreading the array in the max method? We could simple map through and then pass the array to the Math.max function

File: server/controllers/comment.js [hunk 2]
```diff
@@ -104,6 +106,7 @@ class Comment {
     }
 
     const { slug, id } = request.params;
+
     const article = await articleRepo.getArticleBySlug(slug);
     if (article === null) {
```
Comment (victor-nweze-deel) [inline server/controllers/comment.js:111]: We have a checkIfArticleExists middleware. That could be implemented here, so the route gets intercepted if the article doesn't exist

**Review by victor-nweze-deel — COMMENTED** [review]
Please revisit

File: server/controllers/comment.js (hunk 2 above)
Comment (TheoOkafor) [reply server/controllers/comment.js:111]: Oh. Thank you.

File: server/controllers/comment.js (hunk 1 above)
Comment (TheoOkafor) [reply server/controllers/comment.js:135]: Math.max takes numeric parameters. The spread is required in order to parse the parentIds as parameters to the Math.max method.

**Comment by PriscillaSam** [comment]
Great job done @TheoOkafor. Awesome. 👍

**Comment by ucheg6** [comment]
Great Job @TheoOkafor. This looks great.

**Comment by seyi-adeleke** [comment]
Your pr title is wrong

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]
...

**Comment by TheoOkafor** [comment]
Fixed. Thank you

**Comment by TheoOkafor** [comment]
> Your pr title is wrong

Fixed. Thank you

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**